### PR TITLE
Support push preview docs.

### PR DESCRIPTION
**Pull request by fredrikekre**
Documenter can now push preview docs from PRs (when the head branch is from the same repository). This is enabled by passing `push_preview=true` to `deploydocs`. Documenter will then push rendered documentation to `previews/PR##` where `##` is the pull request number.

### CODE REVIEW

**Comment by fredrikekre** [comment]
See https://github.com/fredrikekre/Literate.jl/pull/80

**Comment by fredrikekre** [comment]
- [x] Push from GH Actions: https://github.com/fredrikekre/Literate.jl/commit/0effc2e2974433640116db60a6aedc6483da4406 (although the page build is not triggered...)
- [x] Push from Travis: https://github.com/fredrikekre/Literate.jl/commit/a79843c2252aaae8b23a23c29a8e4b01074172bc

Resulting page: https://fredrikekre.github.io/Literate.jl/previews/PR80/

**Comment by fredrikekre** [comment]
https://juliadocs.github.io/Documenter.jl/previews/PR1180/ :tada:

File: src/Documenter.jl [hunk 1]
```diff
@@ -747,8 +747,7 @@ function doctest(
             )
             true
         catch err
-            @error "Doctesting failed"
-            showerror(stdout, err, catch_backtrace())
+            @error "Doctesting failed" exception=(err, catch_backtrace())
```
Comment (mortenpi) [inline src/Documenter.jl]: Didn't know that this works :+1:

**Pull request merged**